File: csharp/darts/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y) =>
        dist(x, y) switch
        {
            <= 1.0 => 10,
            <= 5.0 => 5,
            <= 10.0 => 1,
            _ => 0
        };

    private static double dist(double x, double y) => Math.Sqrt(Math.Pow(x, 2) + Math.Pow(y, 2));
}

File: erlang/darts/src/darts.erl
-module(darts).

-export([score/2]).

dist(X,Y) -> math:sqrt(X * X + Y * Y).

score(_X, _Y) ->
  D = dist(_X, _Y),
  if
    D =< 1 -> 10;
    D =< 5 -> 5;
    D =< 10 -> 1;
    true -> 0
  end.
